promql/parser/lex.go

```diff
@@ -217,7 +220,7 @@ type Pos int
 // Lexer holds the state of the scanner.
 type Lexer struct {
 	input       string  // The string being scanned.
-	state       stateFn // The next lexing function to enter.
+	State       stateFn // The next lexing function to enter.
```

Comment:
`State` doesn't need to be exported, either.

Reply:
Basically almost all of the changes to this file should be rolled back.

Reply:
Oh yes, this went unnoticed.

Reply:
BTW, `State` is used inside the `lexItems()` in `prettier.go`.

Reply:
Sth. like

```go
lexer := parser.Lex(query)

for item := lexer.NextItem(); item.Typ != parser.EOF; item = lexer.NextItem {
   ...
}
```

Reply:
Done. Resolving.